engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1832,7 +1832,7 @@ private Pair<VolumeVO, DataStore> recreateVolume(VolumeVO vol, VirtualMachinePro
                 break; //break out of template-redeploy retry loop
             } catch (StorageAccessException e) {
                 throw e;
-            } catch (InterruptedException | ExecutionException e) {
+            } catch (InterruptedException | ExecutionException | NullPointerException e) {
```

Comment:
for test purposes ; let the NPE pass through
```suggestion
            } catch (InterruptedException | ExecutionException e) {
```